workflow/controller/controller.go

```diff
@@ -878,12 +878,6 @@ func (wfc *WorkflowController) processNextItem(ctx context.Context) bool {
 	startTime := time.Now()
 	woc.operate(ctx)
 	wfc.metrics.OperationCompleted(time.Since(startTime).Seconds())
-	if woc.wf.Status.Fulfilled() {
-		err := woc.completeTaskSet(ctx)
-		if err != nil {
-			log.WithError(err).Warn("error to complete the taskset")
-		}
-	}
```

Comment:
Sorry, can you explain why this is no longer needed? If I am correct, it's because the persistUpdates will handle this, correct?